lib/internal/quic/core.js

```diff
@@ -0,0 +1,2882 @@
+      ...options
+    };
+    const {
+      alpn,
+      dcid,
+      ipv6Only,
+      minDHSize,
+      port,
+      preferredAddressPolicy,
+      remoteTransportParams,
+      requestOCSP,
+      servername,
+      sessionTicket,
+      verifyHostnameIdentity,
+      qlog,
+    } = validateQuicClientSessionOptions(options);
+
+    if (!verifyHostnameIdentity && !warnedVerifyHostnameIdentity) {
+      warnedVerifyHostnameIdentity = true;
+      process.emitWarning(
+        'QUIC hostname identity verification is disabled. This violates QUIC ' +
+        'specification requirements and reduces security. Hostname identity ' +
+        'verification should only be disabled for debugging purposes.'
+      );
+    }
+
+    super(socket, servername, alpn);
+    this.#dcid = dcid;
+    this.#ipv6Only = ipv6Only;
+    this.#minDHSize = minDHSize;
+    this.#port = port || 0;
+    this.#preferredAddressPolicy = preferredAddressPolicy;
+    this.#remoteTransportParams = remoteTransportParams;
+    this.#requestOCSP = requestOCSP;
+    this.#secureContext =
+      createSecureContext(
+        sc_options,
+        initSecureContextClient);
+    this.#sessionTicket = sessionTicket;
+    this.#transportParams = validateTransportParams(options);
+    this.#verifyHostnameIdentity = verifyHostnameIdentity;
+    this.#qlogEnabled = qlog;
+  }
+
+  [kHandshakePost]() {
+    const { type, size } = this.ephemeralKeyInfo;
+    if (type === 'DH' && size < this.#minDHSize) {
+      this.destroy(new ERR_TLS_DH_PARAM_SIZE(size));
+      return false;
+    }
+
+    // TODO(@jasnell): QUIC *requires* that the client verify the
+    // identity of the server so we'll need to do that here.
+    // The current implementation of tls.checkServerIdentity is
+    // less than great and could be rewritten to speed it up
+    // significantly by running at the C++ layer. As it is
+    // currently, the method pulls the peer cert data, converts
+    // it to a javascript object, then processes the javascript
+    // object... which is more expensive than what is strictly
+    // necessary.
+    //
+    // See: _tls_wrap.js onConnectSecure function
+
+    return true;
+  }
+
+  [kContinueConnect](type, ip) {
+    const flags = this.#ipv6Only ? UV_UDP_IPV6ONLY : 0;
+    setTransportParams(this.#transportParams);
+
+    const options =
+      (this.#verifyHostnameIdentity ?
+        QUICCLIENTSESSION_OPTION_VERIFY_HOSTNAME_IDENTITY : 0) |
+      (this.#requestOCSP ?
+        QUICCLIENTSESSION_OPTION_REQUEST_OCSP : 0);
+
+    const handle =
+      _createClientSession(
+        this.socket[kHandle],
+        type,
+        ip,
+        this.#port,
+        flags,
+        this.#secureContext.context,
+        this.servername || ip,
+        this.#remoteTransportParams,
+        this.#sessionTicket,
+        this.#dcid,
+        this.#preferredAddressPolicy,
+        this.alpnProtocol,
+        options,
+        this.#qlogEnabled);
+    // We no longer need these, unset them so
+ 
```

Comment:
`autoDestroy: true` (or at least explicit `false`)?

Reply:
@jasnell: FYI, master now sets `autoDestroy: true` by default... in case that affects anything.